### PR TITLE
FluentUI Widgets Localization Support

### PR DESCRIPTION
**Pull request by henry2man**
This PR adds support for onboard widgets localization support using `intl` package.

Related with https://github.com/bdlukaa/fluent_ui/issues/213

I've included a wide list of most spoken languages: ~~Arabic, Bengali, English, Spanish, French, Hindi, Indonesian, Portuguese, Russian and Simplified Chinese using Google Translate.~~ **UPDATE**: ⭐️ @bdlukaa and I (@henry2man) agreed that we prefer confirmed localizations vs "Google Translate" ones, so we need some help!

# How to contribute:

- Get any of the already verified translation files like https://github.com/bdlukaa/fluent_ui/pull/216/files#diff-d38a1072799c0e0b8936fb54ea8bcc732c76241afed954d025815eb8369aad32R1-R27
- Get the file you want to review: for example: https://github.com/bdlukaa/fluent_ui/pull/216/files#diff-613ee02ebb6ae3d3300afd6f00fe30255b8e37336c5d89170bbbf9cee108c1daR1-R27
- Review and copy the contents of your reviewed file and leave a comment!
- Then we will merge them into the branch 👍🏻

Current status:

- [x] English ✅
- [X] Spanish ✅ - Verified by @henry2man 
- [X] French ✅ - Already verified by @WinXaito
- [x] Portuguese ✅ - Reviewed by @bdlukaa 
- [x] Russian ✅ - Reviewed by @raitonoberu
- [x] German ✅ - Reviewed by @Larsb24
- [x] Hindi ✅ - Reviewed by @alexmercerind
- [x] Simplified Chinese ✅ - Reviewed by @zacksleo
- Other languages are welcomed!

I've added detailed instructions for adding new localizations in the `README.md`:  https://github.com/bdlukaa/fluent_ui/pull/216/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

-----------------------------

If you run the example app and change the language of the browser, the corresponding messages are shown.

Also I've done some little tweaks to shortcuts label implementation. AFAIK they aren't strictly that needs a translation (`Ctrl` or `Cmd` are standards names for corresponding keys). 

Please take notice of changes here: https://github.com/bdlukaa/fluent_ui/pull/216/files#diff-3705042816f3bd9d24304192811bed9d5104fa569a36f063a4166cfea97b845aR376-R377 and here https://github.com/bdlukaa/fluent_ui/pull/216/files#diff-8b7e9df87668ffa6a04b32e1769a33434999e54ae081c52e5d943c541d4c0d25R13-R14 in order to support multilanguage material localizations.

Lastly, take a look at this `List<Locale>` constant: https://github.com/bdlukaa/fluent_ui/pull/216/files#diff-ccec5c5fa2c4d7bf6bc0e9748a8836c4bc5ff4a8c59461448d14047f6b242617R128-R139

As the note says, this would be defined into `DefaultFluentLocalizations` but I cannot be able to make it work, so this was a working workarround.

## Pre-launch Checklist

- [x] I have run `dartfmt` on all changed files
- [x] I have updated `CHANGELOG.md` with my changes  
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could
- [x] I have added/updated relevant documentation
- [ ] I have run `flutter pub publish --dry-run` and addressed any warnings

### CODE REVIEW

File: lib/l10n/intl_fr.arb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "restoreWindowTooltip": "Restaurer",
+  "closeWindowTooltip": "Fermer",
+  "dialogLabel": "Dialogue",
+  "cutActionLabel": "Couper",
+  "copyActionLabel": "Copier",
+  "pasteActionLabel": "Coller",
+  "selectAllActionLabel": "Tout sélectionner",
+  "newTabLabel": "Ajouter un nouvel onglet",
+  "closeTabLabelSuffix": "Fermer l'onglet",
+  "scrollTabBackwardLabel": "Défilement vers l'arrière",
```
Comment (WinXaito) [inline lib/l10n/intl_fr.arb:20]: `Défiler vers l'arrière` to match with the next item `Défiler vers l'avant`

**Comment by WinXaito** [comment]
Awesome ! :smile:

File: lib/l10n/intl_fr.arb (hunk 1 above)
Comment (henry2man) [reply lib/l10n/intl_fr.arb]: @WinXaito Solved!

**Comment by henry2man** [comment]
You can consider this PR finished, so it can be merged if you like it.

**Comment by bdlukaa** [comment]
Were these generated by the translator? I don't like the translator idea. For example, the Portuguese translation is totally incorrect

**Comment by bdlukaa** [comment]
> In this cases I always prefer a bad translation vs no-translation at all, but this is something that is very easy to fix with this new implementation.

It's better a confirmed translation IMO

**Comment by henry2man** [comment]
> I'll check on my discord server (https://discord.gg/674gpDQUVq)

Sounds great!

File: lib/l10n/intl_pt.arb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "closeButtonLabel": "Fechar",
+  "searchLabel": "Pesquisar",
+  "closeNavigationTooltip": "Fechar navegador",
+  "openNavigationTooltip": "Abrir navegador",
+  "clickToSearch": "Clique para pesquisar",
+  "modalBarrierDismissLabel": "Cancelar",
+  "minimizeWindowTooltip": "Minimizar",
+  "restoreWindowTooltip": "Restaurar",
+  "closeWindowTooltip": "Fechar",
+  "dialogLabel": "Diálogo",
```
Comment (bdlukaa) [inline lib/l10n/intl_pt.arb:13]:
```suggestion
  "dialogLabel": "Caixa de diálogo",
```

File: lib/l10n/intl_pt.arb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+  "@@locale": "pt",
+  "backButtonTooltip": "Voltar",
+  "closeButtonLabel": "Fechar",
+  "searchLabel": "Pesquisar",
+  "closeNavigationTooltip": "Fechar navegador",
+  "openNavigationTooltip": "Abrir navegador",
```
Comment (bdlukaa) [inline lib/l10n/intl_pt.arb:7]:
```suggestion
  "closeNavigationTooltip": "Fechar navegação",
  "openNavigationTooltip": "Abrir navegação",
```

File: lib/l10n/intl_pt.arb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  "selectAllActionLabel": "Selecionar tudo",
+  "newTabLabel": "Adicionar nova guia",
+  "closeTabLabelSuffix": "Fechar guia",
+  "scrollTabBackwardLabel": "Rolar para trás",
+  "scrollTabForwardLabel": "Rolar para frente",
+  "noResultsFoundLabel": "Nenhum resultado encontrado",
+  "copyActionTooltip": "Copiar conteúdo selecionado para a área de transferência",
+  "cutActionTooltip": "Recortar o conteúdo selecionado e colocá-lo na área de transferência",
+  "pasteActionTooltip": "Colar o conteúdo da área de transferência na posição atual",
+  "selectAllActionTooltip": "Selecionar todo o conteúdo"
```
Comment (bdlukaa) [inline lib/l10n/intl_pt.arb:26]:
```suggestion
  "selectAllActionTooltip": "Selecionar tudo"
```

**Review by bdlukaa — CHANGES_REQUESTED** [review]
Translator isn't that bad tho

**Comment by raitonoberu** [comment]
`lib/l10n/intl_ru.arb`
<details>

```json
{
  "@@locale": "ru",
  "backButtonTooltip": "Назад",
  "closeButtonLabel": "Закрыть",
  "searchLabel": "Поиск",
  "closeNavigationTooltip": "Закрыть панель навигации",
  "openNavigationTooltip": "Открыть панель навигации",
  "clickToSearch": "Нажмите для поиска",
  "modalBarrierDismissLabel": "Отмена",
  "minimizeWindowTooltip": "Свернуть",
  "restoreWindowTooltip": "Восстановить",
  "closeWindowTooltip": "Закрыть",
  "dialogLabel": "Диалог",
  "cutActionLabel": "Вырезать",
  "copyActionLabel": "Копировать",
  "pasteActionLabel": "Вставить",
  "selectAllActionLabel": "Выбрать все",
  "newTabLabel": "Новая вкладка",
  "closeTabLabelSuffix": "Закрыть вкладку",
  "scrollTabBackwardLabel": "Прокрутить назад",
  "scrollTabForwardLabel": "Прокрутить вперед",
  "noResultsFoundLabel": "Результаты не найдены",
  "copyActionTooltip": "Скопировать в буфер обмена",
  "cutActionTooltip": "Вырезать и поместить в буфер обмена",
  "pasteActionTooltip": "Вставить содержимое буфера обмена",
  "selectAllActionTooltip": "Выбрать все"
}
```
</details>

File: lib/l10n/intl_fr.arb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  "pasteActionLabel": "Coller",
+  "selectAllActionLabel": "Tout sélectionner",
+  "newTabLabel": "Ajouter un nouvel onglet",
+  "closeTabLabelSuffix": "Fermer l'onglet",
+  "scrollTabBackwardLabel": "Défiler vers l'arrière",
+  "scrollTabForwardLabel": "Défiler vers l'avant",
+  "noResultsFoundLabel": "Aucun résultat trouvé",
+  "copyActionTooltip": "Copier le contenu sélectionné dans le presse-papiers",
+  "cutActionTooltip": "Couper le contenu sélectionné et le mettre dans le presse-papiers",
+  "pasteActionTooltip": "Coller le contenu du presse-papiers à la position actuelle",
```
Comment (WinXaito) [inline lib/l10n/intl_fr.arb:25]:
```suggestion
  "copyActionTooltip": "Copier le contenu sélectionné dans le presse-papier",
  "cutActionTooltip": "Couper le contenu sélectionné et le mettre dans le presse-papier",
  "pasteActionTooltip": "Coller le contenu du presse-papier à la position actuelle",
```

Sorry, I didn't see it the first time

Comment (WinXaito) [reply lib/l10n/intl_fr.arb:25]: Btw, I checked and both versions are syntactically correct (thank you French...) So leave it like that.

Comment (WinXaito) [reply lib/l10n/intl_fr.arb:25]: Btw 2, I see on wikipedia that `presse-papiers` with the **s** refers to the old spelling. So you can still apply the change

Comment (henry2man) [reply lib/l10n/intl_fr.arb:25]: Well, if there is a newer spelling we can adopt it! Thanks @WinXaito

**Review by alexmercerind — CHANGES_REQUESTED** [review]
So, I translated/reviewed the Hindi language.

Thanks. 
```json
{
  "@@locale": "hi",
  "backButtonTooltip": "वापस",
  "closeButtonLabel": "बंद करें",
  "searchLabel": "खोजें",
  "closeNavigationTooltip": "नेविगेशन बंद करें",
  "openNavigationTooltip": "नेविगेशन खोलें",
  "clickToSearch": "खोजने के लिए क्लिक करें",
  "modalBarrierDismissLabel": "हटाएँ",
  "minimizeWindowTooltip": "मिनीमाइज करें",
  "restoreWindowTooltip": "वापिस लाएं",
  "closeWindowTooltip": "बंद करें",
  "dialogLabel": "डायलॉग",
  "cutActionLabel": "कट",
  "copyActionLabel": "कॉपी",
  "pasteActionLabel": "पेस्ट",
  "selectAllActionLabel": "सब-कुछ सेलेक्ट करें",
  "newTabLabel": "नया टैब ऐड करें",
  "closeTabLabelSuffix": "टैब बंद करें",
  "scrollTabBackwardLabel": "टैब लिस्ट पीछे स्क्रॉल करें",
  "scrollTabForwardLabel": "टैब लिस्ट आगे स्क्रॉल करें",
  "noResultsFoundLabel": "कोई रिजल्ट नहीं मिला",
  "copyActionTooltip": "सेलेक्टेड कंटेंट क्लिपबोर्ड पर कॉपी करें",
  "cutActionTooltip": "सिलेक्टेड कंटेंट यहाँ से हटा कर क्लिपबोर्ड पर कॉपी करें",
  "pasteActionTooltip": "क्लिपबोर्ड का कंटेंट इस लोकेशन पर पेस्ट करें",
  "selectAllActionTooltip": "सारा कंटेंट सेलेक्ट करें"
}
```

**Comment by henry2man** [comment]
> So, I translated/reviewed the Hindi language.

@alexmercerind Merged 👏🏻 thanks!!

**Comment by zacksleo** [comment]
I've reviewed Simplified Chinese, thanks!

``` json
{
  "@@locale": "zh",
  "backButtonTooltip": "返回",
  "closeButtonLabel": "关闭",
  "searchLabel": "搜索",
  "closeNavigationTooltip": "关闭导航",
  "openNavigationTooltip": "打开导航",
  "clickToSearch": "点击搜索",
  "modalBarrierDismissLabel": "取消",
  "minimizeWindowTooltip": "最小化",
  "restoreWindowTooltip": "恢复",
  "closeWindowTooltip": "关闭",
  "dialogLabel": "对话",
  "cutActionLabel": "剪切",
  "copyActionLabel": "复制",
  "pasteActionLabel": "粘贴",
  "selectAllActionLabel": "全选",
  "newTabLabel": "添加新标签",
  "closeTabLabelSuffix": "关闭标签",
  "scrollTabBackwardLabel": "向后滚动标签列表",
  "scrollTabForwardLabel": "向前滚动标签列表",
  "noResultsFoundLabel": "没有找到结果",
  "copyActionTooltip": "将选中的内容复制到剪贴板",
  "cutActionTooltip": "将选中的内容剪切到剪贴板",
  "pasteActionTooltip": "在当前位置插入剪贴板的内容",
  "selectAllActionTooltip": "选择所有内容"
}
```
```flutter
  static Map<String, Function> _notInlinedMessages(_) => <String, Function>{
        "backButtonTooltip": MessageLookupByLibrary.simpleMessage("返回"),
        "clickToSearch": MessageLookupByLibrary.simpleMessage("点击搜索"),
        "closeButtonLabel": MessageLookupByLibrary.simpleMessage("关闭"),
        "closeNavigationTooltip": MessageLookupByLibrary.simpleMessage("关闭导航"),
        "closeTabLabelSuffix": MessageLookupByLibrary.simpleMessage("关闭标签"),
        "closeWindowTooltip": MessageLookupByLibrary.simpleMessage("关闭"),
        "copyActionLabel": MessageLookupByLibrary.simpleMessage("复制"),
        "copyActionTooltip":
            MessageLookupByLibrary.simpleMessage("将选中的内容复制到剪贴板"),
        "cutActionLabel": MessageLookupByLibrary.simpleMessage("剪切"),
        "cutActionTooltip":
            MessageLookupByLibrary.simpleMessage("将选中的内容剪切到剪贴板"),
        "dialogLabel": MessageLookupByLibrary.simpleMessage("对话"),
        "minimizeWindowTooltip": MessageLookupByLibrary.simpleMessage("最小化"),
        "modalBarrierDismissLabel": MessageLookupByLibrary.simpleMessage("取消"),
        "newTabLabel": MessageLookupByLibrary.simpleMessage("添加新标签"),
        "noResultsFoundLabel": MessageLookupByLibrary.simpleMessage("没有找到结果"),
        "openNavigationTooltip": MessageLookupByLibrary.simpleMessage("打开导航"),
        "pasteActionLabel": MessageLookupByLibrary.simpleMessage("粘贴"),
        "pasteActionTooltip":
            MessageLookupByLibrary.simpleMessage("在当前位置插入剪贴板的内容"),
        "restoreWindowTooltip": MessageLookupByLibrary.simpleMessage("恢复"),
        "scrollTabBackwardLabel":
            MessageLookupByLibrary.simpleMessage("向后滚动标签列表"),
        "scrollTabForwardLabel":
            MessageLookupByLibrary.simpleMessage("向前滚动标签列表"),
        "searchLabel": MessageLookupByLibrary.simpleMessage("搜索"),
        "selectAllActionLabel": MessageLookupByLibrary.simpleMessage("全选"),
        "selectAllActionTooltip": MessageLookupByLibrary.simpleMessage("选择所有内容")
      };
```

**Comment by henry2man** [comment]
> I've reviewed Simplified Chinese, thanks!

Merged @zacksleo, you're welcomed! 🚀

PS: We only need the contents of the `intl_xxx.arb` files, the other ones are autogenerated.

**Comment by henry2man** [comment]
I've added detailed instructions in `README.md` so anyone will be capable of contributing new languages and localizations :)

**Comment by henry2man** [comment]
I'm just waiting the German translation for remove `draft` mark for this PR.

**Comment by bdlukaa** [comment]
@henry2man can this be marked as ready for review?

**Comment by henry2man** [comment]
> @henry2man can this be marked as ready for review?

I'm still waiting 🤷🏻‍♂️ for https://github.com/bdlukaa/fluent_ui/pull/216#issuecomment-1065067849

**Comment by bdlukaa** [comment]
yup. sounds good!

File: lib/l10n/intl_de.arb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  "clickToSearch": "Zum Suchen klicken",
+  "modalBarrierDismissLabel": "Schließen",
+  "minimizeWindowTooltip": "Minimieren",
+  "restoreWindowTooltip": "Wiederherstellen",
+  "closeWindowTooltip": "Schließen",
+  "dialogLabel": "Dialog",
+  "cutActionLabel": "Ausschneiden",
+  "copyActionLabel": "Kopieren",
+  "pasteActionLabel": "Einfügen",
+  "selectAllActionLabel": "Alle auswählen",
```
Comment (larsb24) [inline lib/l10n/intl_de.arb:17]: "selectAllActionLabel": "Alles auswählen",

Comment (larsb24) [inline lib/l10n/intl_de.arb:17]: "selectAllActionLabel": "Alles auswählen",

**Comment by henry2man** [comment]
@larsb24 Thanks for your German translation.

@bdlukaa Now it's ready to review & merge 👍🏻

**Review by bdlukaa — APPROVED** [review]

**Comment by dmakwt** [comment]
I translated the Arabic language.
`lib/l10n/intl_ar.arb`

```
{
  "@@locale": "ar",
  "backButtonTooltip": "رجوع",
  "closeButtonLabel": "إغلاق",
  "searchLabel": "بحث",
  "closeNavigationTooltip": "إغلاق الواجهة",
  "openNavigationTooltip": "فتح الواجهة",
  "clickToSearch": "انقر للبحث",
  "modalBarrierDismissLabel": "استبعاد",
  "minimizeWindowTooltip": "تصغير",
  "restoreWindowTooltip": "إسترجاع",
  "closeWindowTooltip": "إغلاق",
  "dialogLabel": "مربع حوار",
  "cutActionLabel": "قص",
  "copyActionLabel": "نسخ",
  "pasteActionLabel": "لصق",
  "selectAllActionLabel": "تحديد الكل",
  "newTabLabel": "إضافة علامة تبويب جديدة",
  "closeTabLabelSuffix": "إغلاق علامة التبويب",
  "scrollTabBackwardLabel": "تمرير قائمة علامة التبويب للخلف",
  "scrollTabForwardLabel": "تمرير قائمة علامة التبويب إلى الأمام",
  "noResultsFoundLabel": "لم يتم العثور على نتائج",
  "copyActionTooltip": "انسخ المحتوى المحدد إلى الحافظة",
  "cutActionTooltip": "قم بإزالة المحتوى المحدد وضعه في الحافظة",
  "pasteActionTooltip": "إدراج محتويات الحافظة إلى الموقع الحالي",
  "selectAllActionTooltip": "تحديد المحتوى بالكامل"
}
```

**Comment by henry2man** [comment]
@dmakwt Great! 🥇Can you create a new pull request like @dubh3 did in https://github.com/bdlukaa/fluent_ui/pull/267?

Please follow the instructions here: https://github.com/bdlukaa/fluent_ui#contributing-new-localizations

**Comment by dmakwt** [comment]
@henry2man yes sure 👍 #268

**Comment by Dyrkon** [comment]
Hi, I have done a translation into Czech.

{
  "@@locale": "cs",
  "backButtonTooltip": "Zpět",
  "closeButtonLabel": "Zavřít",
  "searchLabel": "Hledat",
  "closeNavigationTooltip": "Zavřít navigaci",
  "openNavigationTooltip": "Otevřít navigaci",
  "clickToSearch": "Kliknout pro vyhledávání",
  "modalBarrierDismissLabel": "Zamítnout",
  "minimizeWindowTooltip": "Minimalizovat",
  "restoreWindowTooltip": "Obnovit",
  "closeWindowTooltip": "Zavřít",
  "dialogLabel": "Zpráva",
  "cutActionLabel": "Vyjmout",
  "copyActionLabel": "Kopírovat",
  "pasteActionLabel": "Vložit",
  "selectAllActionLabel": "Vybrat vše",
  "newTabLabel": "Přidat novou záložku",
  "closeTabLabelSuffix": "Zavřít záložku",
  "scrollTabBackwardLabel": "Posunout záložky zpět",
  "scrollTabForwardLabel": "Posunout záložky kupředu",
  "noResultsFoundLabel": "Nebyl nalezen žádný výsledek",
  "copyActionTooltip": "Zkopírovat vybraný obsah do schránky",
  "cutActionTooltip": "Vyjmout vybraný obsah a vložit do schránky",
  "pasteActionTooltip": "Vložit obsah schránky na současné místo",
  "selectAllActionTooltip": "Vybrat všechen obsah"
}

**Pull request merged**